app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/splash/SplashScreen.kt

```diff
@@ -1,58 +1,207 @@
 import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalDensity
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.IntOffset
 import androidx.compose.ui.unit.dp
-import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
 import com.kks.nimblesurveyjetpackcompose.R
-import com.kks.nimblesurveyjetpackcompose.ui.presentation.destinations.LoginScreenDestination
+import com.kks.nimblesurveyjetpackcompose.ui.theme.Concord
+import com.kks.nimblesurveyjetpackcompose.ui.theme.White18
+import com.kks.nimblesurveyjetpackcompose.util.extensions.loginTextFieldModifier
 import com.kks.nimblesurveyjetpackcompose.viewmodel.splash.SplashViewModel
 import com.ramcosta.composedestinations.annotation.Destination
-import com.ramcosta.composedestinations.navigation.DestinationsNavigator
-import com.ramcosta.composedestinations.navigation.EmptyDestinationsNavigator
+import kotlinx.coroutines.launch
 
 @Destination(start = true)
 @Composable
-fun SplashScreen(navigator: DestinationsNavigator, viewModel: SplashViewModel = viewModel()) {
+fun SplashScreen(viewModel: SplashViewModel = hiltViewModel()) {
+    val showLoginComponents by viewModel.shouldNavigateToLogin.collectAsState()
+    val logoOffset = remember { Animatable(Offset(0f, 0f), Offset.VectorConverter) }
+    val positionToAnimate = -LocalDensity.current.run { 221.dp.toPx() }
+
     LaunchedEffect(
-        key1 = viewModel.shouldNavigateToLogin.value
+        key1 = showLoginComponents
     ) {
-        if (viewModel.shouldNavigateToLogin.value) {
-            navigator.navigate(LoginScreenDestination)
+        if (showLoginComponents) {
+            launch {
+                logoOffset.animateTo(
+                    Offset(logoOffset.value.x, positionToAnimate),
+                    spring(
+                        dampingRatio = Spring.DampingRatioLowBouncy,
+                        stiffness = StiffnessVeryLow
+                    )
+                )
+            }
         }
     }
-
     Box(
         modifier = Modifier.fillMaxSize(),
         contentAlignment = Alignment.Center
     ) {
         Image(
-            painter = painterResource(id = R.drawable.splash_bg),
+            painter = painterResource(id = if (showLoginComponents) R.drawable.ic_overlay else R.drawable.splash_bg),
             contentDescription = stringResource(id = R.string.splash_background_content_description),
-            modifier = Modifier.matchParentSize(),
+            modifier = Modifier
+                .matchParentSize(),
```

Comment:
This `Modifier.matchParentSize()` doesn't want to stay on the same line for some reason. 😆